Description: Fix Sage's use of scipy rtol parameter
 The scipy documentation very clearly says: 
 "The parameter cannot be smaller than its default value of
 ``4*np.finfo(float).eps``" so use that instead of a hard-coded value.
Author: Ximin Luo <infinity0@debian.org>
Forwarded: Bug: https://trac.sagemath.org/ticket/22439
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/sage/src/sage/numerical/optimize.py
+++ b/sage/src/sage/numerical/optimize.py
@@ -15,8 +15,11 @@
 from sage.modules.free_module_element import vector
 from sage.rings.real_double import RDF
 
+from numpy import finfo
+default_rtol = 4*finfo(float).eps
 
-def find_root(f, a, b, xtol=10e-13, rtol=4.5e-16, maxiter=100, full_output=False):
+
+def find_root(f, a, b, xtol=10e-13, rtol=default_rtol, maxiter=100, full_output=False):
     """
     Numerically find a root of ``f`` on the closed interval `[a,b]`
     (or `[b,a]`) if possible, where ``f`` is a function in the one variable.
--- a/sage/src/sage/symbolic/expression.pyx
+++ b/sage/src/sage/symbolic/expression.pyx
@@ -11012,7 +11012,7 @@
             ret = ret[0]
         return ret
 
-    def find_root(self, a, b, var=None, xtol=10e-13, rtol=4.5e-16, maxiter=100, full_output=False):
+    def find_root(self, a, b, var=None, xtol=10e-13, rtol=0, maxiter=100, full_output=False):
         """
         Numerically find a root of self on the closed interval [a,b] (or
         [b,a]) if possible, where self is a function in the one variable.
@@ -11128,7 +11128,7 @@
         """
         if is_a_relational(self._gobj) and self.operator() is not operator.eq:
             raise ValueError("Symbolic equation must be an equality.")
-        from sage.numerical.optimize import find_root
+        from sage.numerical.optimize import find_root, default_rtol
         if self.number_of_arguments() == 0:
             if bool(self == 0):
                 return a
@@ -11137,7 +11137,7 @@
         elif self.number_of_arguments() == 1:
             f = self._fast_float_(self.default_variable())
             return find_root(f, a=a, b=b, xtol=xtol,
-                             rtol=rtol,maxiter=maxiter,
+                             rtol=(rtol or default_rtol),maxiter=maxiter,
                              full_output=full_output)
         else:
             raise NotImplementedError("root finding currently only implemented in 1 dimension.")
